### PR TITLE
Add FXIOS-8220 [Component Library] Close button

### PR DESCRIPTION
**Pull request by cyndichin**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8220)

## :bulb: Description
Proposal to add Close Button to Component Library
- Add Close Button component to library
- Updated Close Button in BottomSheet in the library to use component
- Updated Fakespot Button to use the component as an example

**Background Context**
Essentially, after working on microsurvey, I realize the close button is reusable so touched base with Crystal and got a thumbs up on her side in adding to Component Library. 

Other areas using close button:
- Enhanced Tracking Protection Menu
- Micro Survey
- Onboarding - Default Browser, Intro, Update

Note: If team agrees, need to create contributor tickets and update docs .

Ref: https://www.figma.com/file/YIbYarab5aYte7KbK0bcVw/iOS-Components?type=design&node-id=1252-2029&mode=design&t=ZPwkYgQTFFM9KVTj-0

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

# Screensh
| Button | BottomSheet |
| --- | --- |
| Simulator Screenshot - iPhone 15 Plus - 2024-04-26 at 14 46 22 | Simulator Screenshot - iPhone 15 Plus - 2024-04-26 at 14 46 13 |

| Before | After |
| --- | --- |
| simulator_screenshot_2DEFCE5F-6F34-4BF8-8BBD-7D7ED63E01BE | simulator_screenshot_56256030-DAA3-4CAA-B6B9-E1B56E282C9E |

### CODE REVIEW

**Comment by cyndichin** [comment]
Along with this proposal, updated close button image as requested per @cwzilla

**Review by cwzilla — APPROVED** [review]

File: SampleComponentLibraryApp/SampleComponentLibraryApp/Buttons/ButtonsViewController.swift
Comment (mattreaganmozilla) [inline SampleComponentLibraryApp/SampleComponentLibraryApp/Buttons/ButtonsViewController.swift:64]: Is this a placeholder string?

File: BrowserKit/Sources/ComponentLibrary/Buttons/CloseButton.swift
Comment (mattreaganmozilla) [inline BrowserKit/Sources/ComponentLibrary/Buttons/CloseButton.swift:30]: Nit: if the button is going to own the responsibility of applying its own constraints for a fixed width/size then it is probably a good idea for it to also update its `translatesAutoresizingMaskIntoConstraints` to false. Otherwise problems might occur if someone doesn't construct this button subclass using `.build` (or initializes it directly but forgets to set the above property).

**Review by mattreaganmozilla — COMMENTED** [review]
> The PR itself is TBD since I want to get buy in from the team first.

@cyndichin Changes LGTM; were you just looking for initial feedback or were you planning to merge this once approved?

**Comment by cyndichin** [comment]
> > The PR itself is TBD since I want to get buy in from the team first.
> 
> @cyndichin Changes LGTM; were you just looking for initial feedback or were you planning to merge this once approved?

I got the buy in from the team in this morning meeting, so planning on merging once approved!

File: SampleComponentLibraryApp/SampleComponentLibraryApp/Buttons/ButtonsViewController.swift
Comment (cyndichin) [reply SampleComponentLibraryApp/SampleComponentLibraryApp/Buttons/ButtonsViewController.swift:64]: yes, since this is in the sample component library, I just added some arbitrary accessibility string, happy to change to one that works better!

File: BrowserKit/Sources/ComponentLibrary/Buttons/CloseButton.swift
Comment (cyndichin) [reply BrowserKit/Sources/ComponentLibrary/Buttons/CloseButton.swift:31]: good call! I think this is a good safety check. We should be using `.build` but its not mandated, so thank you for pointing this out!

**Review by mattreaganmozilla — APPROVED** [review]
🚀

File: BrowserKit/Sources/ComponentLibrary/BottomSheet/BottomSheetViewController.swift
Comment (thatswinnie) [inline BrowserKit/Sources/ComponentLibrary/BottomSheet/BottomSheetViewController.swift:93]: I think the a11y identifier should come from the viewModel as well and be added to the `AccessibilityIdentifiers` struct.

File: BrowserKit/Sources/ComponentLibrary/Buttons/CloseButton.swift
Comment (thatswinnie) [inline BrowserKit/Sources/ComponentLibrary/Buttons/CloseButton.swift:5]: nit: import of Foundation is not necessary if you import UIKit.

File: BrowserKit/Sources/ComponentLibrary/Buttons/CloseButton.swift
Comment (thatswinnie) [inline BrowserKit/Sources/ComponentLibrary/Buttons/CloseButton.swift:31]: nit: using self is not necessary here.

**Review by thatswinnie — COMMENTED** [review]
Should the image for dark mode be updated too? I find the contrast pretty low as well, @cwzilla 
Simulator Screenshot - iPhone 15 Pro Max - 2024-04-26 at 09 46 58

File: BrowserKit/Sources/ComponentLibrary/Buttons/CloseButton.swift
Comment (cyndichin) [reply BrowserKit/Sources/ComponentLibrary/Buttons/CloseButton.swift]: thanks for the catch! updated!

File: BrowserKit/Sources/ComponentLibrary/BottomSheet/BottomSheetViewController.swift
Comment (thatswinnie) [reply BrowserKit/Sources/ComponentLibrary/BottomSheet/BottomSheetViewController.swift:93]: Sounds good. Let's do that. Thanks @cyndichin 🙏

**Review by thatswinnie — APPROVED** [review]

**Comment by cwzilla** [comment]
@thatswinnie Thanks for catching the contrast issue! I shared with the a11y team and made some modifications to both the light and dark images. I think previously there was some opacity in the icon which caused it to have low contrast on dark mode. Here are the new files (these will be uploaded to the acorn repo shortly – just having some issues right now). Thanks!
[crossCircleFillExtraLargeDark.pdf](https://github.com/mozilla-mobile/firefox-ios/files/15134247/crossCircleFillExtraLargeDark.pdf)
[crossCircleFillExtraLargeLight.pdf](https://github.com/mozilla-mobile/firefox-ios/files/15134248/crossCircleFillExtraLargeLight.pdf)

**Comment by cyndichin** [comment]
@cwzilla thanks for the assets, I uploaded the new ones and update the screenshots!

**Review by cwzilla — APPROVED** [review]
@cyndichin looks good, thank you so much!!

**Pull request merged**
